newsrevenuehub/rev-engine
# DEV-2717: Org Settings page

Please fill out each section even if it's just with "N/A"

#### Plan? (if this draft/incomplete indicate what you intend to do and how)
This is the first PR of stacked PRs:

1. Org settings page (this PR)
2. Link into AvatarMenu: https://github.com/newsrevenuehub/rev-engine/pull/993
3. Enhancement (Warn message) - https://github.com/newsrevenuehub/rev-engine/pull/996
4. Handle Fiscal Sponsor in Org Settings - https://github.com/newsrevenuehub/rev-engine/pull/997

#### What's this PR do?
- Creates the Settings Organization page (and all missing subcomponents) for users to view information regarding their organization & revenue program.
- Implement view/edit mode depending on user permission/roles

#### Why are we doing this? How does it help us?
This is the one of the steps to implementing view/edit settings related to the user's account.

#### How should this be manually tested? Please include detailed step-by-step instructions.
Test all 4 scenarios:


----
1)
- Log in with ORG admin user
- Org must have only 1 revenue program
- Check `/settings/organization/`
- All fields should be editable
- When changing values, action buttons appear
- Clicking "save" saves the changes and if you refresh the page it should still show the updated values
- Clicking "undo" reverts any changes and all fields go back to original value

---
2)
- Log in with ORG admin user
- Org with more than 1 revenue program
- Check `/settings/organization/`
- All available fields should be editable
- Disclaimer shows
- Tax status & Tax ID fields don't show
- When changing values, action buttons appear
- Clicking "save" saves the changes and if you refresh the page it should still show the updated values
- Clicking "undo" reverts any changes and all fields go back to original value

---
3)
- Log in with RP admin user
- Org must have only 1 revenue program
- Check `/settings/organization/`
- All fields should be view only

---
4)
- Log in with RP admin user
- Org with more than 1 revenue program
- Check `/settings/organization/`
- All fields should be view only
- Disclaimer shows
- Tax status & Tax ID fields don't show

Check AC in https://news-revenue-hub.atlassian.net/browse/DEV-2717

#### Did this PR make changes to the user interface that may require changes to the user-facing docs?
yes

#### Have automated unit tests been added? If not, why?
yes

#### How should this change be communicated to end users?
Viewing and updating organization details is now possible through the settings Organization page.

#### Are there any smells or added technical debt to note?
no

#### Has this been documented? If so, where?
no

#### What are the relevant tickets? Add a link to any relevant ones.
https://news-revenue-hub.atlassian.net/browse/DEV-2717

#### Do any changes need to be made before deployment to production (adding environment variables, for example)?
no

#### Are there next steps? If so, what? Have tickets been opened for them? List next-step tickets here:
This changes are related to the tickets in these 2 epics:
https://news-revenue-hub.atlassian.net/browse/DEV-3067
https://news-revenue-hub.atlassian.net/browse/DEV-3018
